pkg/ebpf/map_cleaner.go

```diff
@@ -107,85 +105,74 @@ func (mc *MapCleaner) Stop() {
 	})
 }
 
-func (mc *MapCleaner) clean(nowTS int64, shouldClean func(nowTS int64, k, v interface{}) bool) {
-	keySize := int(mc.emap.KeySize())
-	keysToDelete := make([][]byte, 0, 128)
-	totalCount, deletedCount := 0, 0
+func (mc *MapCleaner[K, V]) cleanWithBatches(nowTS int64, shouldClean func(nowTS int64, k K, v V) bool) {
 	now := time.Now()
 
-	entries := mc.emap.Iterate()
-	for entries.Next(mc.keyPtr, mc.valPtr) {
-		totalCount++
-
-		if !shouldClean(nowTS, mc.key, mc.val) {
-			continue
+	mc.keysToDelete = mc.keysToDelete[:0]
+	totalCount, deletedCount := 0, 0
+	var next K
+	var n int
+	for {
```

Comment:
Do we really want to execute this in a loop? I think it's not that unlikely that in a real-world workload this could run  forever which would prevent the `MapCleaner` to hang during termination, no? In this pathological scenario you would also keep passing a stale timestamp to the callback

Reply:
Not sure about running forever / adding timed context here
but instead, we can "halt" if we processed `mc.emap.MaxEntries`
what do you think about that?

Along side with aborting the loop when we got 0 entries from the batch api, it will guarantee we won't be stuck forever

Reply:
sounds reasonable to me 👍